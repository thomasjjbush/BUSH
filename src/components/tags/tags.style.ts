import styled from 'styled-components';
import { StyledProps } from '../../types';

export const StyledTag = styled.span<StyledProps>`
    background-color: ${(props): string => props.theme.colors.brand};
    border-radius: 5px;
    color: white;
    font-family: Roboto Mono;
    font-size: 13px;
    padding: 5px;
    margin: 5px 5px 0 0;
`;

export const StyledTags = styled.div`
    display: flex;
    flex-wrap: wrap;
`;
