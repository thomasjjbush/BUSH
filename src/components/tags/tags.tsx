import React, { FunctionComponent, ReactElement } from 'react';
import { StyledTag, StyledTags } from './tags.style';
import { TagProps } from '../../types';

export const Tags: FunctionComponent<TagProps> = ({ tags }: TagProps): ReactElement => {
    return (
        <StyledTags>
            {tags.map(({ name, slug }) => (
                <StyledTag key={slug}>{name}</StyledTag>
            ))}
        </StyledTags>
    );
};
