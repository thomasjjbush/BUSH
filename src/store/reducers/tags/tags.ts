import { TagActions, Tag } from '../../../types';

export const tagReducer = (state: Tag[] = [], action: { payload: Tag[]; type: TagActions }): Tag[] => {
    switch (action.type) {
        case TagActions.LOADED:
            return [...action.payload];
        default:
            return state;
    }
};
