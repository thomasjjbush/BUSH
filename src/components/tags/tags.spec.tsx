import React from 'react';
import { shallow } from 'enzyme';
import { Tags } from './tags';

describe('Tags', () => {
    it('should match snapshot', () => {
        const tags = [
            { name: 'x', slug: 'x' },
            { name: 'y', slug: 'y' },
            { name: 'z', slug: 'z' },
        ];
        expect(shallow(<Tags tags={tags} />)).toMatchSnapshot();
    });
});
