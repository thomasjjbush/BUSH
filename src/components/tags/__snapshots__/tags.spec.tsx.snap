// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tags should match snapshot 1`] = `
<styled.div
  styles="
    display: flex;
    flex-wrap: wrap;
"
>
  <styled.span
    key="x"
    styles="
    background-color: theme.colors.brand;
    border-radius: 5px;
    color: white;
    font-family: Roboto Mono;
    font-size: 13px;
    padding: 5px;
    margin: 5px 5px 0 0;
"
  >
    x
  </styled.span>
  <styled.span
    key="y"
    styles="
    background-color: theme.colors.brand;
    border-radius: 5px;
    color: white;
    font-family: Roboto Mono;
    font-size: 13px;
    padding: 5px;
    margin: 5px 5px 0 0;
"
  >
    y
  </styled.span>
  <styled.span
    key="z"
    styles="
    background-color: theme.colors.brand;
    border-radius: 5px;
    color: white;
    font-family: Roboto Mono;
    font-size: 13px;
    padding: 5px;
    margin: 5px 5px 0 0;
"
  >
    z
  </styled.span>
</styled.div>
`;
